core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -145,7 +147,7 @@ object SparkSubmit {
       }
     // In all other modes, just run the main class as prepared
     } else {
-      runMain(childArgs, childClasspath, sysProps, childMainClass)
+      runMain(childArgs, childClasspath, sysProps, childMainClass, args)
```

Comment:
Yeah, working on it. I'm trying to figure out why the exceptions are not being printed when I do that (the thing I mentioned to you on IM). Hopefully I'll have an update soon.